flink-cdc-connect/flink-cdc-source-connectors/flink-connector-db2-cdc/src/test/java/org/apache/flink/cdc/connectors/db2/table/Db2ConnectorITCase.java

```diff
@@ -309,6 +309,78 @@ public void testAllTypes() throws Exception {
         cancelJobIfRunning(result);
     }
 
+    @Test
+    public void testPrimaryKeyQuote() throws Exception {
+        initializeDb2Table("corner_case_test", "QUOTE_PRIMARY_KEY");
+        String sourceDDL =
+                String.format(
+                        "CREATE TABLE quote_primary_key_source ("
+                                + " `KEY` INT NOT NULL,"
+                                + " NAME STRING,"
+                                + " DESCRIPTION STRING,"
+                                + " WEIGHT DECIMAL(10,3),"
+                                + " PRIMARY KEY (`KEY`) NOT ENFORCED"
+                                + ") WITH ("
+                                + " 'connector' = 'db2-cdc',"
+                                + " 'hostname' = '%s',"
+                                + " 'port' = '%s',"
+                                + " 'username' = '%s',"
+                                + " 'password' = '%s',"
+                                + " 'database-name' = '%s',"
+                                + " 'table-name' = '%s' ,"
+                                + " 'scan.startup.mode' = 'latest-offset',"
+                                + " 'scan.incremental.snapshot.enabled' = '%s'"
+                                + ")",
+                        DB2_CONTAINER.getHost(),
+                        DB2_CONTAINER.getMappedPort(DB2_PORT),
+                        DB2_CONTAINER.getUsername(),
+                        DB2_CONTAINER.getPassword(),
+                        DB2_CONTAINER.getDatabaseName(),
+                        "DB2INST1.QUOTE_PRIMARY_KEY",
+                        incrementalSnapshot);
+        String sinkDDL =
+                "CREATE TABLE sink "
+                        + " WITH ("
+                        + " 'connector' = 'values',"
+                        + " 'sink-insert-only' = 'false'"
+                        + ") LIKE quote_primary_key_source (EXCLUDING OPTIONS)";
+        tEnv.executeSql(sourceDDL);
+        tEnv.executeSql(sinkDDL);
+
+        // async submit job
+        TableResult result =
+                tEnv.executeSql("INSERT INTO sink SELECT * FROM quote_primary_key_source");
+        // wait for the source startup, we don't have a better way to wait it, use sleep for now
+        do {
+            Thread.sleep(5000L);
+        } while (result.getJobClient().get().getJobStatus().get() != RUNNING);
```

Comment:
This might get stuck when job has already reached terminal state. Maybe copy `MySqlTestUtils#waitForJobStatus`'s implementation?